app/components/FileLoader.js

```diff
@@ -290,6 +342,8 @@ export default class FileLoader extends Component {
           file={file}
           playFile={this.props.playFile}
           gameProfileLoad={this.props.gameProfileLoad}
+          onSelect={this.onSelect}
+          selectedOrdinal={this.state.selections.indexOf(file) + 1}
```

Comment:
This is an unnecessary O(n) operation in a loop, so worst case run time of this would be O(nm). It could be avoided by using a keyed object for selections instead where the key is the full path of the file. That would also make the "add selection" logic better as well.